### PR TITLE
Improve the README.md document

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,81 @@
...
+
+- PD: [PD](https://github.com/pingcap/pd) is the abbreviation for Placement Driver. It is used to manage and schedule the TiKV cluster.
+
+    1. Make sure ​Go​ (version 1.12) is installed.
+    2. Use `make` to build the PD (the binary will be placed in the `bin` directory).
+
+- TiKV: [Building and setting up a development workspace](https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#building-and-setting-up-a-development-workspace)
+- TiDB: [Building TiDB on a local OS/shell environment](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md#building-tidb-on-a-local-osshell-environment)
+
+The `copr-test` integration test will retrieve the source code of the TiDB master branch automatically (but you can specify the
```
Comment (dcalvin) [inline README.md]:
```suggestion
The `copr-test` integration test retrieves the source code of the TiDB master branch automatically, but you can also
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,81 @@
...
+TiDB source code path by `$tidb_src_dir` instead of using remote latest TiDB source code), so you should only prepare
```
Comment (dcalvin) [inline README.md]:
```suggestion
specify a local path by using `$tidb_src_dir`. This  means you should only prepare
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,81 @@
...
+
+The `copr-test` is the integration test for `Coprocessor` of the TiKV. The `Coprocessor` is used to execute TiDB push-down
+executor to improve the database performance. It's important to keep execution result consistent between TiDB and `Coprocessor`.
+This is the purpose of `copr-test`.
+
+The basic principle of `copr-test` is to execute test cases in TiDB standalone (with `mocktikv`) and TiDB cluster (with `Coprocessor`)
+and compare the execution result. We think the TiDB and Coprocessor is consistent if the results in both sides are same, vice versa.
+We use `github.com/pingcap/parser` to parse every file, and iterator the []ast.StmtNode to run every SQL.
+So if there is any error in the test SQL file, the error msg is hard to read. Creating the database and tables in you local database
+and use IDE(such as CLion) to write SQL file as test case is recommended.
```
[the remark this answers is not in the document]
Comment (lonng) [reply README.md]: This part is copied from https://github.com/pingcap/tidb-test/pull/857/files#diff-9f0ba7bf050d89f7f897d1996da662b4R6, @H-ZeX PTAL

File: README.md [hunk 4]
```diff
@@ -1,4 +1,81 @@
...
+
+The `Coprocessor` contains two execution frameworks, `Batch` (vectorized execution engine) and `Non-Batch` in current
+stage (remove the `Non-Batch` version is in plan). So all test cases should be passed in the following execution environment.
+
+1. TiDB with mocktikv
+2. TiDB + TiKV with Non-Batch
```
Comment (lonng) [inline README.md]:
```suggestion
- TiDB + TiKV with Non-Batch
```

File: README.md [hunk 5]
```diff
@@ -1,4 +1,81 @@
...
+3. TiDB + TiKV with Batch
```
Comment (lonng) [inline README.md]:
```suggestion
- TiDB + TiKV with Batch
```

File: README.md [hunk 6]
```diff
@@ -1,4 +1,81 @@
...
+
+### How do I know which expressions will be pushed down?
+
+The `Coprocessor` does not implement all builtin functions (or implemented but not tested fully) which have implemented by TiDB,
+so we can only push down the functions that Coprocessor has implemented and fully tested. Whether a function is pushed down or not
+is determined by the function `canFuncBePushed` in TiDB [tidb-expression/expr_to_pb.go](https://github.com/pingcap/tidb/blob/a090e6be2991bf85b18fcdb096f84d41f4f6bd85/expression/expr_to_pb.go#L303)
+
+We have added a failpoint `PushDownTestSwitcher` in the function `canFuncBePushed` to hijack our customized push down condition.
```
[the remark this answers is not in the document]
Comment (lonng) [reply README.md]: How about adding a link here?

File: README.md [hunk 7]
```diff
@@ -1,4 +1,81 @@
...
+
+1. TiDB with mocktikv
+2. TiDB + TiKV with Non-Batch
+3. TiDB + TiKV with Batch
+
+### How do I know which expressions will be pushed down?
+
+The `Coprocessor` does not implement all builtin functions (or implemented but not tested fully) which have implemented by TiDB,
+so we can only push down the functions that Coprocessor has implemented and fully tested. Whether a function is pushed down or not
+is determined by the function `canFuncBePushed` in TiDB [tidb-expression/expr_to_pb.go](https://github.com/pingcap/tidb/blob/a090e6be2991bf85b18fcdb096f84d41f4f6bd85/expression/expr_to_pb.go#L303)
```
[the remark this answers is not in the document]
Comment (lonng) [reply README.md]: OK

**Comment by lonng** [comment]
@dcalvin PTAL again.

File: README.md [hunk 8]
```diff
@@ -1,4 +1,76 @@
...
+    select *
+    from tb2;
+    SELECT CAST(date AS time), cast(date_2 as date), cast(date_3 as date)
+    from tb2;
+    ```
+
+## How to run a test case in your local environment?
+
+As the integration test for TiKV Coprocessor, `copr-test` requires a minimal requirement to run 
+run in your local environment.
```
Comment (dcalvin) [inline README.md]:
```suggestion
in your local environment.
```

File: README.md [hunk 9]
```diff
@@ -1,4 +1,76 @@
...
+
+- PD: [PD](https://github.com/pingcap/pd) is the abbreviation for Placement Driver. It is used to manage and schedule the TiKV cluster.
+
+    1. Make sure ​Go​ (version 1.12) is installed.
+    2. Use `make` to build PD (the binary will be placed in the `bin` directory).
+
+- TiKV: [Building and setting up a development workspace](https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#building-and-setting-up-a-development-workspace)
+- TiDB: [Building TiDB on a local OS/shell environment](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md#building-tidb-on-a-local-osshell-environment)
+
+`copr-test` integration test retrieves the source code of the TiDB master branch automatically, but you can also
```
Comment (dcalvin) [inline README.md]:
```suggestion
`copr-test` integration test retrieves the source code from the TiDB master branch automatically, but you can also
```

File: README.md [hunk 10]
```diff
@@ -1,4 +1,76 @@
...
+specify a local path by using `$tidb_src_dir`. This  means you should only prepare
```
Comment (dcalvin) [inline README.md]:
```suggestion
specify source code of your selection at a local path by using `$tidb_src_dir`. This  means you should only prepare
```

File: README.md [hunk 11]
```diff
@@ -1,4 +1,76 @@
...
+the build environment for it. The binaries of TiKV and PD should be provided by environment variables (`$pd_bin` and `$tikv_bin`).
```
Comment (dcalvin) [inline README.md]:
```suggestion
the build environment for it. The binaries of TiKV and PD are provided via environment variables (`$pd_bin` and `$tikv_bin`).
```

**Pull request merged**